### PR TITLE
Expand macro condition "Scene item transform": allow trigger on transform changed

### PR DESCRIPTION
**Pull request by Oekn5w**
The scene item transform trigger only had the condition to exactly match a given transformation (or a regex match). I think a trigger when the transform changes gives this condition much more usability.

I have left the regex enabled in that case, although I didn't fully check the code.
Hopefully, this enables a checking if specific JSON attribute has changed.

Tested w/o regex for a single source item.

### CODE REVIEW

File: src/macro-core/macro-condition-scene-transform.cpp [hunk 1]
```diff
@@ -69,25 +130,38 @@ std::string MacroConditionSceneTransform::GetShortDesc() const
...
 	  _scenes(new SceneSelectionWidget(window(), true, false, false, true)),
 	  _sources(new SceneItemSelectionWidget(
 		  parent, true, SceneItemSelectionWidget::Placeholder::ANY)),
+		_types(new QComboBox()),
 	  _getSettings(new QPushButton(obs_module_text(
 		  "AdvSceneSwitcher.condition.sceneTransform.getTransform"))),
 	  _settings(new VariableTextEdit(this)),
 	  _regex(new RegexConfigWidget(parent))
 {
+
```
Comment (WarmUpTill) [inline src/macro-core/macro-condition-scene-transform.cpp:153]: I would prefer that empty line to be removed.

File: src/macro-core/macro-condition-scene-transform.cpp [hunk 2]
```diff
@@ -12,19 +12,78 @@ bool MacroConditionSceneTransform::_registered =
...
+			ret = didTransformOfAnySceneItemChange(items, _previousTransform, _regex, newVariable);
+			break;
+
+		default:
+			return false;
+	}
+
+	if (ret) {
+		SetVariableValue(newVariable);
+	}
```
Comment (WarmUpTill) [inline src/macro-core/macro-condition-scene-transform.cpp:86]: I think the variable value should be set regardless of the condition state.

**Comment by WarmUpTill** [comment]
Thanks a lot for submitting the PR! :)

Do you need help regarding the formatting? (It can be a bit tricky to get right)

A few minor nitpicks:
 * Would you mind capitalizing the first letter of your commit message?
 * (see review comments above)

I will try to test it later once the CI job finishes.
Thanks again for your contribution!

**Comment by WarmUpTill** [comment]
Sorry if I should be misunderstanding something:
Is there any reason for the text input area (and the regex options) to be visible when the "transform changed" condition type is selected?

image

As far as I understand there is not and if so I think they should be hidden to avoid confusing users.
If you want I can implement this or you can do so yourself if you prefer. :)

**Comment by Oekn5w** [comment]
Dang it, I didn't see the formatter in the repo.

And yeah, I thought, that one could check for parts of the whole JSON with regex, but I don't think that's feasible and definitely doesn't work with the current code on second thought, so I'll hide all that stuff.

I'll look to have the commit amended in the next days.

**Comment by Oekn5w** [comment]
Rebased to current master, which should fix the mac workflows

**Comment by WarmUpTill** [comment]
Looks good to me!

The only thing I changed is to add [654284a](https://github.com/WarmUpTill/SceneSwitcher/pull/885/commits/654284af0cfd04f258190ae066c5bc3d666e4c7a) to ensure that variables always have access to the scene item transform data regardless of the condition state.

If you don't plan on adding any further changes I would say this is ready to be merged. :)
Let me if that is the case and I will squash my change and merge.

**Comment by Oekn5w** [comment]
I hadn't figured out how to use the set variable value until just now, but then we should not break out of the for loop to ensure that always the last transform is stored there. Also we can write the variable at the end. I'll edit that accordingly.

**Comment by WarmUpTill** [comment]
Merged - Thanks! :)

**Pull request merged**